quinn-proto/src/varint.rs

```diff
@@ -120,6 +120,14 @@ impl std::convert::TryFrom<usize> for VarInt {
     }
 }
 
+impl std::convert::TryFrom<Duration> for VarInt {
+    type Error = VarIntBoundsExceeded;
+    /// Succeeds iff `x.as_millis()` < 2^62
+    fn try_from(x: Duration) -> Result<Self, Self::Error> {
+        VarInt::try_from(x.as_millis())
```

Comment:
Not sure about how specific this is -- if we ever want a config value that's also naturally expressed in `Duration` but requires μs or ns on the wire, what do we do?

Reply:
Yeah, this gives me pause as well. It also introduces the odd possibility of passing a Duration to fields that aren't times at all. I'd prefer to go with the newtype approach leave the setter fallible. @Matthias247 raised the reasonable concern of a newtype making the API less obvious, but I'm not convinced that's a serious detriment as it's not a heavily-trafficked bit of API.

Reply:
Yup, newtype sounds good to me.

Reply:
Thank you for the review. I'll update this to use a newtype, though might be a day or two before I get to it!